src/probnum/quad/_bayesquad.py

```diff
@@ -8,44 +8,44 @@
 """
 
 import warnings
-from typing import Callable, Optional, Tuple, Union
+from typing import Callable, Optional, Tuple
 
 import numpy as np
 
-from probnum.quad.solvers.bq_state import BQInfo
+from probnum.quad.solvers.bq_state import BQIterInfo
 from probnum.randprocs.kernels import Kernel
 from probnum.randvars import Normal
 from probnum.typing import FloatLike, IntLike
 
-from ._integration_measures import GaussianMeasure, IntegrationMeasure, LebesgueMeasure
+from ._integration_measures import IntegrationMeasure, LebesgueMeasure
+from ._quad_typing import DomainLike, _DomainType
 from .solvers import BayesianQuadrature
 
 
 def bayesquad(
     fun: Callable,
     input_dim: int,
```

Comment:
What you wrote is correct. 

We could add `input_dim` as non-optional argument to `bayesquad_from_data` for some kind of consistency. It seems we cannot make it optional in `bayesquad` for the reason you stated, and adding an optional argument to `baysquad_from_data` that we virtually do not need seems not ideal to me. 

Second option is to leave it as is and change it later in case the issue comes up again.  I am fine with either option. But in case we change it, I would do it in another PR for simplicity.